tests/contest/runtimetest/src/tests.rs

```diff
@@ -545,3 +545,34 @@ pub fn test_io_priority_class(spec: &Spec, io_priority_class: IOPriorityClass) {
         eprintln!("error ioprio_get expected priority {expected_priority:?}, got {priority}")
     }
 }
+
+pub fn validate_masked_paths(spec: &Spec) {
+    let linux = spec.linux().as_ref().unwrap();
+    let masked_paths = match linux.masked_paths() {
+        Some(p) => p,
+        None => {
+            eprintln!("in readonly paths, expected some readonly paths to be set, found none");
+            return;
+        }
+    };
+
+    if masked_paths.is_empty() {
+        return;
+    }
+
+    // TODO when https://github.com/rust-lang/rust/issues/86442 stabilizes,
+    // change manual matching of i32 to e.kind() and match statement
+    for path in masked_paths {
+        if let std::io::Result::Err(e) = test_read_access(path) {
+            let errno = Errno::from_raw(e.raw_os_error().unwrap());
+            if errno == Errno::ENOENT {
+                /* This is expected */
+            } else {
+                eprintln!("in masked paths, error in testing read access for path {path} : {e:?}");
+                return;
+            }
+        } else {
+            /* Expected */
+        }
```

Comment:
Is this expected? For masked path we are expecting the path to NOT be readable inside the container, right? So if this is `Ok`, then that should be error, right?

Reply:
Sorry, you're right. I fixed it. [62e5e1a](https://github.com/containers/youki/pull/2950/commits/62e5e1a1a7bd04ea1a06275661f0325ea75da37f)

Reply:
I added this line but encountered the following error, so I'm currently investigating the cause. As soon as I finish, I'll request a review.
```
# Start group masked_paths
1 / 4 : masked_device_nodes : not ok
	container stderr was not empty, found : in masked paths, error in testing read access for path /masked_device: 8630

2 / 4 : masked_paths : not ok
	container stderr was not empty, found : in masked paths, error in testing read access for path /masked-file: 8630

3 / 4 : masked_rel_paths : ok
4 / 4 : masked_symlinks : ok
# End group masked_paths
```